pyEpiabm/pyEpiabm/sweep/intervention_sweep.py

```diff
@@ -36,22 +36,36 @@ def __init__(self):
         """
         # Implemented interventions and their activity status
         self.intervention_active_status = {}
-        self.intervention_params = Parameters.instance().intervention_params
+        self.intervention_params = Parameters.instance().\
+            intervention_params.copy()
+        self.intervention_dict = {'case_isolation': CaseIsolation,
+                                  'place_closure': PlaceClosure,
+                                  'household_quarantine': HouseholdQuarantine,
+                                  'social_distancing': SocialDistancing,
+                                  'disease_testing': DiseaseTesting,
+                                  'vaccine_params': Vaccination,
+                                  'travel_isolation': TravelIsolation}
 
     def bind_population(self, population):
         self._population = population
-        intervention_dict = {'case_isolation': CaseIsolation,
-                             'place_closure': PlaceClosure,
-                             'household_quarantine': HouseholdQuarantine,
-                             'social_distancing': SocialDistancing,
-                             'disease_testing': DiseaseTesting,
-                             'vaccine_params': Vaccination,
-                             'travel_isolation': TravelIsolation}
-
         for intervention in self.intervention_params.keys():
-            params = self.intervention_params[intervention]
-            self.intervention_active_status[(intervention_dict[intervention](
-                population=self._population, **params))] = False
+            if isinstance(self.intervention_params[intervention],
+                          list):
+                for index in range(len(self.intervention_params[
+                        intervention])):
+                    params = self.intervention_params[intervention][index]
```

Comment:
This construction is generally worth avoiding where possible. In this context, you want to access both the index and the values of this list, so it is worth using [`enumerate()`](https://www.geeksforgeeks.org/enumerate-in-python/).

```suggestion
                for index, params in enumerate(self.intervention_params[
                        intervention]):
```